### ISSUE TITLE
IllegalArgumentException: Undeclared build item class 

### ISSUE THREAD

**Issue by cmoulliard** [quarkiverse/quarkus-argocd#33]
## Issue

When we execute the following command in a quarkus project, not yet published on a git repository, we got this error

```
/Users/cmoullia/code/ch007m/code-with-quarkus $ quarkus argocd application generate                                   
[jbang] [WARN] Detected missing or out-of-date dependencies in cache.
[jbang] Resolving dependencies...
[jbang]    io.quarkiverse.argocd:quarkus-argocd-cli:999-SNAPSHOT
[jbang] Dependencies resolved
java.lang.IllegalArgumentException: Undeclared build item class io.quarkiverse.argocd.spi.ArgoCDApplicationListBuildItem
        at io.quarkus.builder.Messages.undeclaredItem(Messages.java:13)
        at io.quarkus.builder.BuildResult.consume(BuildResult.java:50)
        at io.quarkiverse.argocd.cli.handlers.GetArgoCDApplicationHandler.accept(GetArgoCDApplicationHandler.java:14)
        at io.quarkiverse.argocd.cli.handlers.GetArgoCDApplicationHandler.accept(GetArgoCDApplicationHandler.java:10)
        at io.quarkus.runner.bootstrap.AugmentActionImpl.performCustomBuild(AugmentActionImpl.java:160)
        at io.quarkiverse.argocd.cli.application.GenerateCommand.call(GenerateCommand.java:58)
        at io.quarkiverse.argocd.cli.application.GenerateCommand.call(GenerateCommand.java:25)
```
@iocanel

_Originally posted by @cmoulliard in https://github.com/quarkiverse/quarkus-argocd/issues/24#issuecomment-2370559697_

**Comment by iocanel** [comment]
I can't reporduce

**Comment by cmoulliard** [comment]
This github flow reproduces too the error: https://github.com/quarkiverse/quarkus-argocd/actions/runs/11018349049/job/30598408941 @iocanel

**Comment by cmoulliard** [comment]
Is it possible that when we create here https://github.com/ch007m/quarkus-argocd/blob/main/cli/src/main/java/io/quarkiverse/argocd/cli/application/GenerationBaseCommand.java#L58-L70 the QuarkusBootstrap object that such an object will also use the dependecnies part of the pom.xml as jar/artifacts to its classloader ? @iocanel @geoand

**Comment by geoand** [comment]
I am afraid I don't understand what you are asking with ^.

**Comment by cmoulliard** [comment]
> I am afraid I don't understand what you are asking with ^.

This is pretty simple. Why is the ArgoCD CLI and QuarkusBootstrap working when the following dependency is part of the pom.xml file of the project where we run "quarkus argocd application generate"

```
        <dependency>
            <groupId>io.quarkiverse.argocd</groupId>
            <artifactId>quarkus-argocd</artifactId>
            <version>999-SNAPSHOT</version>
        </dependency>
```

**Comment by cmoulliard** [comment]
Problem is fixed since release: `0.1.0`

Test done successfully
```
❯ quarkus plug add io.quarkiverse.argocd:quarkus-argocd-cli:0.1.0
Added plugin:
    Name         Type    Scope           Location                                                Description    
 *  argocd       maven   project         io.quarkiverse.argocd:quarkus-argocd-cli:0.1.0                         

❯ quarkus argocd project generate .argocd
ArgoCD Resources generated:
 Project                 Application             Repo                                                                    Sync Status     Health Status  
 my-quarkus-hello        my-quarkus-hello        https://gitea.cnoe.localtest.me:8443/giteaAdmin/my-quarkus-hello.git

❯ ls -la .argocd
total 16
drwxr-xr-x@  4 cmoullia  staff  128 Dec 18 10:33 ./
drwxr-xr-x@ 17 cmoullia  staff  544 Dec 18 10:32 ../
-rw-r--r--@  1 cmoullia  staff  659 Dec 18 10:29 application-my-quarkus-hello.yaml
-rw-r--r--@  1 cmoullia  staff  529 Dec 18 10:33 appproject-my-quarkus-hello.yaml
                               
```
